src/main/java/randoop/output/FailingTestFilter.java

```diff
@@ -0,0 +1,235 @@
+       */
+      int totalFailures = -1;
+      Iterator<String> lineIterator = status.standardOutputLines.iterator();
+      while (lineIterator.hasNext() && totalFailures < 0) {
+        String line = lineIterator.next();
+        Matcher messageMatcher = FAILURE_MESSAGE_PATTERN.matcher(line);
+        if (messageMatcher.matches()) {
+          totalFailures = Integer.parseInt(messageMatcher.group(1));
+        }
+      }
+      assert lineIterator.hasNext()
+          : "JUnit has non-zero exit status, but we didn't find a failure";
+
+      /*
+       * Then read the rest of the file to find each failure.
+       * The standard runner gives a numbered list with each entry matching the following pattern:
+       */
+      Pattern failureHeaderPattern =
+          Pattern.compile("\\d+\\)\\s+(" + ID_STRING + ")\\(" + classname + "\\)");
+
+      // Split the class text string so that we can match the line number for the assertion with the code.
+      String[] classLines =
+          classString.split(Globals.lineSep); //use same line break as used to write file
+
+      int failureCount = 0;
+      while (lineIterator.hasNext() && failureCount < totalFailures) {
+        String methodName;
+        int lineNumber;
+        String line = lineIterator.next();
+        Matcher failureMatcher = failureHeaderPattern.matcher(line);
+        if (failureMatcher.matches()) { // found the beginning of a failure
+          failureCount++;
+          methodName = failureMatcher.group(1);
+
+          /*
+           * If the method name in the failure message is not a test method, throw an exception.
+           */
+          if (!methodName.matches(GenTests.TEST_METHOD_NAME_PREFIX + "\\d+")) {
+            if (line.contains("initializationError")) {
+              throw new BugInRandoopException(
+                  "Check configuration of test environment: "
+                      + "initialization error of test in flaky-test filter");
+            } else {
+              throw new BugInRandoopException(
+                  "Unexpected failure in flaky-test filter: " + methodName);
+            }
+          }
+
+          /*
+           * Search for the stacktrace entry corresponding to the test method, and capture the line
+           * number.
+           */
+          Pattern linePattern =
+              Pattern.compile(
+                  "\\s+at\\s+"
+                      + classname
+                      + "\\."
+                      + methodName
+                      + "\\("
+                      + classname
+                      + "\\.java:(\\d+)\\)");
+          lineNumber = -1;
+          while (lineIterator.hasNext() && lineNumber < 0) {
+            line = lineIterator.next();
+            Matcher lineMatcher = linePattern.matcher(line);
+            if (lineMatcher.matches()) {
+              lineNumber = Integer.parseInt(lineMatcher.group(1));
+            }
+          }
```

Comment:
lineNumber might be -1 at this point.  The other 3 while loops checked for the possibility of no match, but this one does not.  Why not?

Also, you could abstract out all the while loops into a readUntilRegexMatch routine that would be called at each of the places that currently has a while loop.  This would simplify and clarify the code, and guarantee consistent error handling.